Client/Frontend/Home/HistoryPanel.swift

```diff
@@ -430,3 +493,125 @@ class HistoryPanel: SiteTableViewController, HomePanel {
+        return chevron
+    }()
+
+    lazy var topBorder: UIView = {
+        let view = UIView()
+        view.backgroundColor = SiteTableViewControllerUX.HeaderBorderColor
+        return view
+    }()
+
+    lazy var bottomBorder: UIView = {
+        let view = UIView()
+        view.backgroundColor = SiteTableViewControllerUX.HeaderBorderColor
+        return view
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        userInteractionEnabled = true
+        backgroundColor = SiteTableViewControllerUX.HeaderBackgroundColor
+
+        let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(SyncedTabsTableViewCell.viewWasTapped(_:)))
+        tapGestureRecognizer.numberOfTapsRequired = 1
+        addGestureRecognizer(tapGestureRecognizer)
+
+        addSubview(topBorder)
+        addSubview(bottomBorder)
+        addSubview(chevron)
+        addSubview(titleLabel)
+        addSubview(imageView)
+
+        imageView.snp_makeConstraints { make in
+            make.left.equalTo(self).offset(TwoLineCellUX.BorderViewMargin + (0.5 * TwoLineCellUX.BorderViewMargin))
+            make.centerY.equalTo(self)
+            make.width.equalTo(HistoryPanelUX.SyncedTabsCellImageSize)
+        }
+
+        chevron.snp_makeConstraints { make in
+            make.right.equalTo(self).offset(-HistoryPanelUX.SyncedTabsCellChevronInset)
+            make.centerY.equalTo(self)
+            make.size.equalTo(HistoryPanelUX.SyncedTabsCellChevronSize)
+        }
+
+        topBorder.snp_makeConstraints { make in
+            make.left.right.equalTo(self)
+            make.top.equalTo(self).offset(-0.5)
+            make.height.equalTo(0.5)
+        }
+
+        bottomBorder.snp_makeConstraints { make in
+            make.left.right.bottom.equalTo(self)
+            make.height.equalTo(0.5)
+        }
+
+        updateConstraints()
+    }
+
+    override func updateConstraints() {
+        super.updateConstraints()
+        descriptionLabel?.removeFromSuperview()
+
+        titleLabel.snp_remakeConstraints { make in
+            make.left.equalTo(imageView.snp_right).offset(TwoLineCellUX.BorderViewMargin)
+            make.centerY.equalTo(self)
+        }
+
+        if let description = descriptionLabel {
+            addSubview(description)
+
+            description.snp_makeConstraints { make in
+                make.left.equalTo(imageView.snp_right).offset(TwoLineCellUX.BorderViewMargin)
+                make.centerY.equalTo(self).offset(10)
+            }
+
+            titleLabel.snp_remakeConstraints { make in
+                make.left.equalTo(imageView.snp_right).offset(TwoLineCellUX.BorderViewMargin)
+                make.centerY.equalTo(self).offset(-10)
+            }
+        }
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    @objc private func viewWasTapped(gestureRecognizer: UITapGestureRecognizer) {
```

Comment:
As a general rule of thumb, try to see `UIView` subclasses as 'logic-less' as possible. Ideally views should contain layout code and that's about it. I would move this tap gesture selector up into the `HistoryPanel` class where you have access to `navigationController` and `historyPanelController` instead of passing them down to the view.